### PR TITLE
linode-cli: enable/add test; add bash auto-completion

### PR DESCRIPTION
**Pull request by superherointj**
###### Motivation for this change

- Enables tests.
- Add a test for version.
- Add bash auto-completion.
- Add myself (superherointj) as maintainer.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/virtualization/linode-cli/default.nix [hunk 1]
```diff
@@ -48,14 +49,22 @@ buildPythonApplication rec {
...
-  doCheck = false;
+  doInstallCheck = true;
+  installCheckPhase = ''
+    $out/bin/linode-cli --skip-config --version | grep ${version} > /dev/null
+  '';
+
+  nativeBuildInputs = [ installShellFiles ];
+  postInstall = ''
+    $out/bin/linode-cli --skip-config completion bash > linode-cli.bash
+    installShellCompletion linode-cli.bash
```
Comment (SuperSandro2000) [inline pkgs/tools/virtualization/linode-cli/default.nix:60]:
```suggestion
    installShellCompletion --bash --name linode-cli.bash <($out/bin/linode-cli --skip-config completion bash)
```

Comment (superherointj) [reply pkgs/tools/virtualization/linode-cli/default.nix]: Thanks.

File: pkgs/tools/virtualization/linode-cli/default.nix [hunk 2]
```diff
@@ -48,14 +49,21 @@ buildPythonApplication rec {
...
-  # requires linode access token for unit tests, and running executable
-  doCheck = false;
+  doInstallCheck = true;
+  installCheckPhase = ''
+    $out/bin/linode-cli --skip-config --version | grep ${version} > /dev/null
+  '';
+
+  nativeBuildInputs = [ installShellFiles ];
+  postInstall = ''
+    installShellCompletion --bash --name linode-cli.bash <($out/bin/linode-cli --skip-config completion bash)
```
Comment (SuperSandro2000) [inline pkgs/tools/virtualization/linode-cli/default.nix:59]:
```suggestion
    installShellCompletion --bash --cmd linode-cli <($out/bin/linode-cli --skip-config completion bash)
```

Comment (superherointj) [reply pkgs/tools/virtualization/linode-cli/default.nix]: Done.

**Pull request merged**